mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,423 @@
+| time_dimension 	                    | Integer 	     | 100	                | dimension of time vector from `time2vec` paper  	|
+| num_edge_features 	                | Integer 	     | 50           	    | number of edge features we will use from each edge	|
+| num_node_features 	                | Integer 	     | 50 	                | number of expected node features	|
+| message_dimension 	                | Integer 	     | 100 	                | dimension of message, only used if you use MLP as message function type, otherwise ignored 	|
+| num_neighbors 	                    | Integer 	     | 15 	                | number of sampled neighbors 	|
+| edge_message_function_type 	        | String 	     | `identity` 	        | message function type, `identity` for concatenation or `mlp` for projection 	|
+| message_aggregator_type 	            | String 	     | `last` 	            | message aggregator type, `mean` or `last` 	|
+| memory_updater_type 	                | String 	     | `gru` 	            | memory updater type, `gru` or `rnn` 	|
+| num_attention_heads 	                | Integer 	     | 1 	                | number of attention heads used if you define `graph_attn` as layer type  	|
+| learning_rate 	                    | Float 	     | 1e-4	                | learning rate for `adam` optimizer  |
+| weight_decay 	                        | Float 	     | 5e-5 	            | weight decay used in `adam` optimizer 	|
+| device_type 	                        | String 	     | `cuda` 	            | type of device you want to use for training -  `cuda` or `cpu` 	|
+| node_features_property 	            | String 	     | `features`	        | name of features property on nodes from which we read features 	|
+| edge_features_property 	            | String 	     | `features` 	        | name of features property on edges from which we read features	|
+| node_label_property 	                | String 	     | `label` 	            |  name of label property on nodes from which we read features	|
+
+#### Usage:
+
+```cypher
+ CALL tgn.set_params({learning_type:'self_supervised', batch_size:200, num_of_layers:2, 
+                      layer_type:'graph_attn',memory_dimension:20, time_dimension:50, 
+                      num_edge_features:20, num_node_features:20, message_dimension:100, 
+                      num_neighbors:15, edge_message_function_type:'identity',
+                      message_aggregator_type:'last', memory_updater_type:'gru', num_attention_heads:1});
+```
+
+
+### `update(edges)`
+This function scrapes data from edge, including `edge_features` and `node_features` if they exist. It also fills up the batch, and once the batch is ready **TGN** will process
+batch, and be ready to accept new incoming edges.
+#### Input:
+
+* `edges: mgp.List[mgp.Edges]` ➡ list of edges that arrive in a stream to `Memgraph` database to preprocess. If a batch is full, `train` or `eval` starts, depending on the mode.
```

Comment:
```suggestion
* `edges: mgp.List[mgp.Edges]` ➡ list of edges to preprocess (that arrive in a stream to Memgraph). If a batch is full, `train` or `eval` starts, depending on the mode.
```
The current wording was ambiguous on who does the preprocessing: TGN or Memgraph.